src/components/dialog/dialog.js

```diff
@@ -336,6 +336,10 @@ function MdDialogDirective($$rAF, $mdTheming, $mdDialog) {
  *   - `targetEvent` - `{DOMClickEvent=}`: A click's event object. When passed in as an option,
  *     the location of the click will be used as the starting point for the opening animation
  *     of the the dialog.
+ *   - `openFrom` - `{string|DOMElement|object}`: The query selector, DOM element or the rect
+ *     that the dialog will open from
```

Comment:
or the Rect object that is used to determine the bounds (top, left, height, width) from which the Dialog will originate.